contracts/Hub.sol

```diff
@@ -9,29 +9,58 @@ contract Hub {
         uint256 requestedAt;
     }
 
-    // todo Should be more than one inheritance process per heir
     mapping (address => Request) public requests;
+    mapping (address => bytes) public registeredKeys;
 
+    event PublicKeyRegistered(address indexed user, bytes publicKey);
     event MessageRequestedToPublish(address indexed heir);
     event MessageDemandNotToPublish(address indexed heir, address indexed requester);
     event MessagePublished(address indexed heir, string message);
+    event EIMSentToOracle(address indexed owner, bytes encryptedData);
+    event EIMRequestedByHeir(address indexed heir);
+    event EIMSentToHeir(address indexed heir, bytes encryptedData);
+
+    function registerPubKeyOwner(bytes calldata publicKey) public {
+        registeredKeys[msg.sender] = publicKey;
+        emit PublicKeyRegistered(msg.sender, publicKey);
+    }
+
+    function registerPubKeyHeir(address heir, bytes calldata publicKey) public {
+        registeredKeys[heir] = publicKey;
+        emit PublicKeyRegistered(heir, publicKey);
+    }
+
+    function registerPubKeyOracle(bytes calldata publicKey) public {
+        registeredKeys[msg.sender] = publicKey;
+        emit PublicKeyRegistered(msg.sender, publicKey);
+    }
```

Comment:
You can use just one `registerPubKey(bytes calldata pubKey)` method.

Reply:
One more points to consider: you can put the check that msg.sender corresponds to address. Take the logic from this code:

```
contract PublicKeyToAddress {
    
    function publicKeyToAddress(bytes memory publicKey) public pure returns (address) {
        // Check that the public key is of the correct length (64 bytes for uncompressed Ethereum public keys)
        require(publicKey.length == 64, "Invalid public key length");
        
        // Compute the address by taking the last 20 bytes of the keccak256 hash of the public key
        bytes32 hash = keccak256(publicKey);
        return address(uint160(uint256(hash)));
    }
}
```